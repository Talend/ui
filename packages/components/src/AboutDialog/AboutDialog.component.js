import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';
import { withTranslation } from 'react-i18next';

import { Dialog, Skeleton, Icon } from '../';
import getDefaultT from '../translate';
import theme from './AboutDialog.scss';

import I18N_DOMAIN_COMPONENTS from '../constants';

function Text({ text, loading, size = Skeleton.SIZES.medium }) {
	return <div>{loading ? <Skeleton type={Skeleton.TYPES.text} size={size} /> : text}</div>;
}

function Table({ services, loading, t }) {
	if (!services || !services.length) {
		return null;
	}

	return (
		<table className={classNames(theme['about-versions'], 'about-versions')}>
			<thead>
				<tr>
					<th>{t('SERVICE', { defaultValue: 'Service' })}</th>
					<th>{t('BUILD_ID', { defaultValue: 'Build ID' })}</th>
					<th>{t('VERSION', { defaultValue: 'Version' })}</th>
				</tr>
			</thead>
			<tbody>
				{(loading
					? [{ name: 'loading-first' }, { name: 'loading-second' }, { name: 'loading-third' }]
					: services
				).map(service => (
					<tr key={service.name}>
						<td>
							<Text loading={loading} text={service.name} />
						</td>
						<td>
							<Text loading={loading} text={service.build} />
						</td>
						<td>
							<Text loading={loading} text={service.version} />
						</td>
					</tr>
				))}
			</tbody>
		</table>
	);
}

function AboutDialog({
	services,
	expanded,
	show,
	product,
	version,
	loading,
	icon,
	copyrights,
	onToggle,
	onHide,
	t,
}) {
	const bar = {
		actions: {
			center: [
				{
					label: expanded
						? t('LESS', { defaultValue: 'Less' })
						: t('MORE', { defaultValue: 'More' }),
					className: 'btn-default btn-inverse',
					onClick: onToggle,
				},
			],
		},
	};

	return (
		<Dialog
			header={t('ABOUT_HEADER', { defaultValue: 'About {{product}}', product })}
			className={classNames(theme['about-dialog'], 'about-dialog')}
			type={Dialog.TYPES.INFORMATIVE}
			onHide={onHide}
			actionbar={bar}
			show={show}
		>
			<div>
				<Icon name={icon} className={classNames(theme['about-logo'], 'about-logo')} />
				<div className={classNames(theme['about-excerpt'], 'about-excerpt')}>
					<Text
						text={t('ABOUT_VERSION_NAME', {
							defaultValue: 'Version: {{version}}',
							version,
							interpolation: { escapeValue: false },
						})}
						size={Skeleton.SIZES.xlarge}
						loading={loading}
					/>
					<Text
						text={
							copyrights ||
							t('ABOUT_COPYRIGHTS', { defaultValue: 'Â© 2018 Talend. All Rights Reserved' })
						}
						size={Skeleton.SIZES.large}
						loading={loading}
					/>
				</div>
				{expanded && <Table t={t} loading={loading} services={services} />}
			</div>
		</Dialog>
	);
}

AboutDialog.displayName = 'AboutDialog';
AboutDialog.defaultProps = {
	t: getDefaultT(),
};

if (process.env.NODE_ENV !== 'production') {
	Text.propTypes = {
		text: PropTypes.string,
		loading: PropTypes.bool,
		size: PropTypes.string,
	};

	Table.propTypes = {
		services: PropTypes.arrayOf(
			PropTypes.shape({
				name: PropTypes.string,
				version: PropTypes.string,
				build: PropTypes.string,
			}),
		),
		t: PropTypes.func.isRequired,
	};

	AboutDialog.propTypes = {
		expanded: PropTypes.bool,
		show: PropTypes.bool,
		loading: PropTypes.bool,
		copyrights: PropTypes.string,
		onToggle: PropTypes.func,
		onHide: PropTypes.func,
		product: PropTypes.string,
		version: PropTypes.string,
		icon: PropTypes.string,
		t: PropTypes.func.isRequired,
		...Table.propTypes,
	};

	AboutDialog.defaultProps = {
		expanded: false,
		show: false,
		loading: false,
		copyright: null,
		version: '',
		icon: '',
		product: 'this product',
		t: getDefaultT(),
	};
}

export default withTranslation(I18N_DOMAIN_COMPONENTS)(AboutDialog);
