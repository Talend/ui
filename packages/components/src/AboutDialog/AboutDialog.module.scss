@use '~@talend/bootstrap-theme/src/theme/guidelines' as *;

.about-dialog {
	:global(.modal-body) {
		text-align: center;
		padding: {
			top: 0.625rem;
			left: 2rem;
			right: 2rem;
		}

		.about {
			&-logo {
				width: 4.125rem;
				height: 4.125rem;
			}

			&-excerpt {
				margin-top: 1.25rem;
			}

			&-versions {
				margin: 0 auto;
				margin-top: 1.875rem;
				text-align: justify;
				table-layout: fixed;

				th,
				td {
					padding: $padding-smaller;
					text-align: left;
				}
			}
		}
	}
}
