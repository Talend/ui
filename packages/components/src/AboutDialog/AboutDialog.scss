.about-dialog {
	:global(.modal-body) {
		text-align: center;
		padding: {
			top: 1rem;
			left: 3.2rem;
			right: 3.2rem;
		}

		.about {
			&-logo {
				width: 6.6rem;
				height: 6.6rem;
			}

			&-excerpt {
				margin-top: 2rem;
			}

			&-versions {
				margin: 0 auto;
				margin-top: 3rem;
				text-align: justify;

				th {
					text-transform: uppercase;
				}

				td + td,
				th + th {
					border-left: 8rem solid transparent;
				}
			}
		}
	}
}
