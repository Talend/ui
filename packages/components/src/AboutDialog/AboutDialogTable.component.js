import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

import { getI18nInstance } from '../translate';
import Skeleton from '../Skeleton';

import theme from './AboutDialog.scss';

const i18n = getI18nInstance();

export const getColumnHeaders = () => ({
	name: { key: 'name', label: i18n.t('tui-components:SERVICE', { defaultValue: 'Service' }) },
	build: { key: 'build', label: i18n.t('tui-components:BUILD_ID', { defaultValue: 'Build ID' }) },
	version: { key: 'version', label: i18n.t('tui-components:VERSION', { defaultValue: 'Version' }) },
});

export function Text({ text, loading, size = Skeleton.SIZES.medium }) {
	return <div>{loading ? <Skeleton type={Skeleton.TYPES.text} size={size} /> : text}</div>;
}

export function AboutDialogTable({
	definition = Object.values(getColumnHeaders()),
	services = [],
	loading = false,
}) {
	if (!services || !services.length) {
		return null;
	}

	return (
		<table className={classNames(theme['about-versions'], 'about-versions', 'table table-striped')}>
			<thead>
				<tr>
					{definition.map(attribute => (
						<th key={attribute.key}>{attribute.label}</th>
					))}
				</tr>
			</thead>
			<tbody>
				{(loading
					? [{ name: 'loading-first' }, { name: 'loading-second' }, { name: 'loading-third' }]
					: services
				).map(service => (
					<tr key={service.name}>
						{definition.map(attribute => (
							<td key={attribute.key}>
								<Text loading={loading} text={service[attribute.key]} />
							</td>
						))}
					</tr>
				))}
			</tbody>
		</table>
	);
}

Text.propTypes = {
	text: PropTypes.string,
	loading: PropTypes.bool,
	size: PropTypes.string,
};

AboutDialogTable.propTypes = {
	services: PropTypes.arrayOf(
		PropTypes.shape({
			name: PropTypes.string,
			version: PropTypes.string,
			build: PropTypes.string,
		}),
	),
	definition: PropTypes.arrayOf(
		PropTypes.shape({
			key: PropTypes.string,
			label: PropTypes.string,
		}),
	),
	loading: PropTypes.bool,
};
