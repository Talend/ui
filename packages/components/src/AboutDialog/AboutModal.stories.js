import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import AboutDialog from '.';

const props = {
	show: true,
	onToggle: action('onToggle'),
	version: "Summer '18",
	icon: 'talend-tdp-colored',
	services: ['API', 'Dataset', 'Preparation', 'Transformation'].map(name => ({
		version: '2.8.0-SNAPSHOT',
		build: '87d0dcd-12e0d6f',
		name,
	})),
};

const services = [
	'API',
	'Dataset',
	'Preparation',
	'Transformation',
	'service2',
	'service3',
	'service4',
	'service5',
	'service6',
	'service7',
	'service8',
	'service9',
	'service12',
	'service13',
	'service14',
	'service15',
	'service16',
	'service17',
	'service18',
	'service19',
].map(name => ({
	version: '2.8.0-SNAPSHOT',
	build: '87d0dcd-12e0d6f',
	name,
}));

const { name, version } = AboutDialog.Table.getColumnHeaders();

storiesOf('Layout/Modals/AboutModal', module)
	.addDecorator(story => (
		<div>
			<h1>AboutDialog</h1>
			{story()}
		</div>
	))
	.add('default', () => <AboutDialog {...props} />)
	.add('without the version', () => <AboutDialog {...props} version={null} />)
	.add('loading', () => <AboutDialog loading {...props} />)
	.add('expanded', () => <AboutDialog expanded {...props} />)
	.add('expanded with lot of services', () => (
		<AboutDialog expanded {...props} services={services} />
	))
	.add('with custom definition', () => (
		<AboutDialog expanded {...props} services={services} definition={[name, version]} />
	))
	.add('expanded & loading', () => <AboutDialog expanded loading {...props} />);
