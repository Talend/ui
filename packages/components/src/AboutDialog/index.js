import AboutDialog from './AboutDialog.component';

export default AboutDialog;
