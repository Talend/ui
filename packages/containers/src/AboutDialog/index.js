import AboutDialog from './AboutDialog.connect';

import AboutDialogSagas from './AboutDialog.sagas';

AboutDialog.sagas = AboutDialogSagas;

export default AboutDialog;
