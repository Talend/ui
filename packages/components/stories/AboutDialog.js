import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { checkA11y } from '@storybook/addon-a11y';
import { AboutDialog, IconsProvider } from '../src/index';

const props = {
	show: true,
	onToggle: action('onToggle'),
	version: 'Summer \'18',
	icon: 'talend-tdp-colored',
	services: ['API', 'Dataset', 'Preparation', 'Transformation'].map(name => ({
		version: '2.8.0-SNAPSHOT',
		build: '87d0dcd-12e0d6f',
		name,
	})),
};


storiesOf('AboutDialog', module)
	.addDecorator(checkA11y)
	.addDecorator(story => (
		<div>
			<IconsProvider />
			<h1>AboutDialog</h1>
			{story()}
		</div>
	))
	.addWithInfo('default', () => <AboutDialog {...props} />)
	.addWithInfo('loading', () => <AboutDialog loading {...props} />)
	.addWithInfo('expanded', () => <AboutDialog expanded {...props} />)
	.addWithInfo('expanded & loading', () => <AboutDialog expanded loading {...props} />);
