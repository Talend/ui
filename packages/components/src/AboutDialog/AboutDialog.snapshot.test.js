import React from 'react';
import { shallow } from 'enzyme';
import AboutDialog from './AboutDialog.component';

jest.mock('react-dom');

const props = {
	show: true,
	onToggle: jest.fn(),
	version: 'Summer 18',
	icon: 'talend-tdp-colored',
	services: ['API', 'Dataset', 'Preparation', 'Transformation'].map(name => ({
		version: '2.8.0-SNAPSHOT',
		build: '87d0dcd-12e0d6f',
		name,
	})),
};

describe('AboutDialog', () => {
	beforeAll(() => {
		const RealDate = Date;
		global.Date = jest.fn(() => new RealDate(2018, 1, 1));
	});

	it('should render', () => {
		const wrapper = shallow(<AboutDialog.WrappedComponent {...props} />);
		expect(wrapper.getElement()).toMatchSnapshot();
	});

	it('should render with custom copyright', () => {
		const wrapper = shallow(
			<AboutDialog.WrappedComponent copyrights="custom copyrights" {...props} />,
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});

	it('should render services', () => {
		const wrapper = shallow(<AboutDialog.WrappedComponent {...props} expanded />);
		expect(wrapper.getElement()).toMatchSnapshot();
	});

	it('should render in loading mode', () => {
		const wrapper = shallow(<AboutDialog.WrappedComponent {...props} loading />);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
