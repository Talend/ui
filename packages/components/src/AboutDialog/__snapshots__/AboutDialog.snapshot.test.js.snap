// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutDialog should render 1`] = `
<Dialog
  actionbar={
    Object {
      "actions": Object {
        "center": Array [
          Object {
            "className": "btn-default btn-inverse",
            "label": "More",
            "onClick": [MockFunction],
          },
        ],
      },
    }
  }
  autoFocus={true}
  backdrop={true}
  className="theme-about-dialog about-dialog"
  closeButton={true}
  enforceFocus={true}
  header="About this product"
  keyboard={true}
  restoreFocus={true}
  show={true}
  type="informative"
>
  <div>
    <Icon
      className="theme-about-logo about-logo"
      name="talend-tdp-colored"
    />
    <div
      className="theme-about-excerpt about-excerpt"
    >
      <Text
        loading={false}
        size="xlarge"
        text="Version: Summer 18"
      />
      <Text
        loading={false}
        size="large"
        text="© 2018 Talend. All Rights Reserved"
      />
    </div>
  </div>
</Dialog>
`;

exports[`AboutDialog should render in loading mode 1`] = `
<Dialog
  actionbar={
    Object {
      "actions": Object {
        "center": Array [
          Object {
            "className": "btn-default btn-inverse",
            "label": "More",
            "onClick": [MockFunction],
          },
        ],
      },
    }
  }
  autoFocus={true}
  backdrop={true}
  className="theme-about-dialog about-dialog"
  closeButton={true}
  enforceFocus={true}
  header="About this product"
  keyboard={true}
  restoreFocus={true}
  show={true}
  type="informative"
>
  <div>
    <Icon
      className="theme-about-logo about-logo"
      name="talend-tdp-colored"
    />
    <div
      className="theme-about-excerpt about-excerpt"
    >
      <Text
        loading={true}
        size="xlarge"
        text="Version: Summer 18"
      />
      <Text
        loading={true}
        size="large"
        text="© 2018 Talend. All Rights Reserved"
      />
    </div>
  </div>
</Dialog>
`;

exports[`AboutDialog should render services 1`] = `
<Dialog
  actionbar={
    Object {
      "actions": Object {
        "center": Array [
          Object {
            "className": "btn-default btn-inverse",
            "label": "Less",
            "onClick": [MockFunction],
          },
        ],
      },
    }
  }
  autoFocus={true}
  backdrop={true}
  className="theme-about-dialog about-dialog"
  closeButton={true}
  enforceFocus={true}
  header="About this product"
  keyboard={true}
  restoreFocus={true}
  show={true}
  type="informative"
>
  <div>
    <Icon
      className="theme-about-logo about-logo"
      name="talend-tdp-colored"
    />
    <div
      className="theme-about-excerpt about-excerpt"
    >
      <Text
        loading={false}
        size="xlarge"
        text="Version: Summer 18"
      />
      <Text
        loading={false}
        size="large"
        text="© 2018 Talend. All Rights Reserved"
      />
    </div>
    <Table
      loading={false}
      services={
        Array [
          Object {
            "build": "87d0dcd-12e0d6f",
            "name": "API",
            "version": "2.8.0-SNAPSHOT",
          },
          Object {
            "build": "87d0dcd-12e0d6f",
            "name": "Dataset",
            "version": "2.8.0-SNAPSHOT",
          },
          Object {
            "build": "87d0dcd-12e0d6f",
            "name": "Preparation",
            "version": "2.8.0-SNAPSHOT",
          },
          Object {
            "build": "87d0dcd-12e0d6f",
            "name": "Transformation",
            "version": "2.8.0-SNAPSHOT",
          },
        ]
      }
      t={[Function]}
    />
  </div>
</Dialog>
`;

exports[`AboutDialog should render with custom copyright 1`] = `
<Dialog
  actionbar={
    Object {
      "actions": Object {
        "center": Array [
          Object {
            "className": "btn-default btn-inverse",
            "label": "More",
            "onClick": [MockFunction],
          },
        ],
      },
    }
  }
  autoFocus={true}
  backdrop={true}
  className="theme-about-dialog about-dialog"
  closeButton={true}
  enforceFocus={true}
  header="About this product"
  keyboard={true}
  restoreFocus={true}
  show={true}
  type="informative"
>
  <div>
    <Icon
      className="theme-about-logo about-logo"
      name="talend-tdp-colored"
    />
    <div
      className="theme-about-excerpt about-excerpt"
    >
      <Text
        loading={false}
        size="xlarge"
        text="Version: Summer 18"
      />
      <Text
        loading={false}
        size="large"
        text="custom copyrights"
      />
    </div>
  </div>
</Dialog>
`;
